contracts/zero-ex/contracts/src/features/MetaTransactionsFeature.sol

```diff
@@ -453,6 +457,81 @@ contract MetaTransactionsFeature is
         );
     }
 
+    /// @dev Execute a `INativeOrdersFeature.fillLimitOrder()` meta-transaction call
+    ///      by decoding the call args and translating the call to the internal
+    ///      `INativeOrdersFeature._fillLimitOrder()` variant, where we can override
+    ///      the taker address.
+    function _executeFillLimitOrderCall(ExecuteState memory state)
+        private
+        returns (bytes memory returnResult)
+    {
+        LibNativeOrder.LimitOrder memory order;
+        LibSignature.Signature memory signature;
+        uint128 takerTokenFillAmount;
+
+        bytes memory data = state.mtx.callData;
+        bytes memory args = new bytes(data.length - 4);
+        uint256 from;
+        uint256 to;
+        assembly {
+            from := add(data, 36) // skip length and selector
+            to := add(args, 32)   // after length
+        }
+        LibBytesV06.memCopy(to, from, data.length - 4);
+
+        (order, signature, takerTokenFillAmount) = abi.decode(args, (LibNativeOrder.LimitOrder, LibSignature.Signature, uint128));
+
+        return _callSelf(
+            state.hash,
+            abi.encodeWithSelector(
```

Comment:
Actually... I wonder if if we could just modify in-place the `mtx.state.callData`. This is untested, but something like:
```
assembly {
    // length on input calldata
    let mtxCallDataLen := mload(data)

    // we'll extend the input calldata with two extra fields (0x40)
    mstore(data, add(mtxCallDataLen, 0x40)) 

    // where in memory to append these fields (right after state.mtx.callData)
    let mtxCallDataEnd := add(data, add(0x20, mul(mtxCallDataLen, 0x20))) 

    // add fields to end of state.mtx.callData
    mstore(mtxCallDataEnd, taker)
    mstore(add(mtxCallDataEnd, 0x20), sender)
}
```

Reply:
^^ we'd also have to replace the selector w/ `INativeOrdersFeature._fillLimitOrder.selector`

Reply:
IIUC, this is a bit dangerous. We'd be writing to memory that may not be safe to touch. How important is gas efficiency here?

Reply:
I guess we do control the struct, so we can make sure we have room to overwrite things. But I think we still need to decide how important gas efficiency is before we do that sort of thing.

Reply:
Yeah... I was thinking we could probably show it's safe so long as we don't add code after executing the MTX. But, looking more thoroughly, I see we could also be running in a `batchExecute` in which case it's probably pretty likely we're overwriting someone else's memory.